cpp/src/phonenumbers/phonenumbermatcher.cc

```diff
@@ -656,6 +656,16 @@ bool PhoneNumberMatcher::Next(PhoneNumberMatch* match) {
 bool PhoneNumberMatcher::Find(int index, PhoneNumberMatch* match) {
   DCHECK(match);
 
+  // Input should should contain only UTF-8 characters.
```

Comment:
I don't think this is the right place to check this. That means you do this check every time you call next.
Can we do it in the constructor? The text does not change. And store it as a boolean, and if it's false, return false from HasNext?

Reply:
Hi Lara,

I have not blocked PhoneNumberMatcher creation as issues can be returned only through returning exceptions in constructor. As adding CPP exceptions are not encouraged (also introduce backward incompatibility -compile errors), lazily returning no matches (in Find() API), similar to earlier commits.

Yes, I would have explored alternate options like:
- doing this check in constructor and flag it to reference later during Find()/any as mentioned in https://stackoverflow.com/a/45643989
- I do not see other options that are innocuous. Please suggest.

Reply:
Doing this check in constructor & adding a boolean flag sounds reasonable to me, to avoid checking the entire input text repeatedly.

Reply:
Indeed :)